src/server/systems/NetworkingServer.cpp

```diff
@@ -168,23 +174,41 @@ void NetworkingServer::handleUnreliableCommand(sf::Socket::Status status, sf::Pa
             break;
         }
         default: {
-            std::cout << "Unknown command code sent: " << unreliable_command.command << std::endl;
+            std::cerr << "Unknown command code sent: " << unreliable_command.command << std::endl;
             break;
         }
     }
 }
 
 void NetworkingServer::setClientUnreliableUpdate(sf::Packet packet, int client_id,
                                                  uint client_tick) {
+    sf::Uint32 player_id;
+    {
+        std::lock_guard<std::mutex> m_clientToPlayerIds_guard(m_clientToPlayerIds_lock);
+        if (m_clientToPlayerIds_t.count(client_id) == 0) {
+            return;
+        }
+        player_id = m_clientToPlayerIds_t[client_id];
+    }
+
     int drift = std::abs(static_cast<int>((m_tick_ta - client_tick)));
+    {
+        std::lock_guard<std::mutex> m_playerIdsToDriftMetric_guard(m_playerIdsToDriftMetrics_lock);
+        if (m_playerIdsToDriftMetrics_t.find(player_id) == m_playerIdsToDriftMetrics_t.end()) {
+            m_playerIdsToDriftMetrics_t[player_id] = new TemporalMetric{120, sf::seconds(0.5f)};
+        }
+        m_playerIdsToDriftMetrics_t[player_id]->pushCount(drift);
+        m_playerIdsToDriftMetrics_t[player_id]->update();
+
+    }
```

Comment:
maybe worth moving this to its own method, the networking code is already a bit tough to read so it would be nice if you could just read one line for all of this and think "oh and we log a drift metric here"